packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerModule.kt

```diff
@@ -0,0 +1,75 @@
+package expo.modules.taskManager
+
+import android.os.Handler
+import expo.modules.core.errors.ModuleNotFoundException
+import expo.modules.interfaces.taskManager.TaskManagerInterface
+import expo.modules.interfaces.taskManager.TaskServiceInterface
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+
+class TaskManagerModule : Module() {
+  private val taskService: TaskServiceInterface? by lazy {
+    appContext.legacyModuleRegistry.getSingletonModule("TaskService", TaskServiceInterface::class.java)
+  }
+  private val taskManagerInternal: TaskManagerInterface? by lazy {
+    appContext.legacyModule()
+  }
+
```

Comment:
What do you think about adding another property here?
```kotlin
private val taskService get() = ensuresTaskService()
````
 Rename the interface to something like this
```kotlin
private val taskServiceInterface: TaskServiceInterface? by lazy {
  appContext.legacyModuleRegistry.getSingletonModule("TaskService", TaskServiceInterface::class.java)
}
```
 and remove the calls to `ensuresTaskService()` in each function?